library/core/src/pin.rs

```diff
@@ -485,6 +485,16 @@ impl<P: Deref<Target: Unpin>> Pin<P> {
     ///
     /// Unlike `Pin::new_unchecked`, this method is safe because the pointer
     /// `P` dereferences to an [`Unpin`] type, which cancels the pinning guarantees.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::pin::Pin;
+    ///
+    /// let val: u8 = 5;
+    /// // We can pin the value, since it doesn't care about being moved
+    /// let pinned: Pin<&u8> = Pin::new(&val);
```

Comment:
It seems like an example of Pin with a shared reference is probably not the best bet -- most users will not want that. I think we should demonstrate with a &mut reference at least.

Reply:
```rust
    let mut val: u8 = 5;
    let mut pinned: Pin<&mut u8> = Pin::new(&mut val);
    println!("{}", pinned); // 5
    pinned.as_mut().set(10);
    println!("{}", pinned); // 10
```
Would this work?